basic/atm/atm_exercise.py

```diff
@@ -0,0 +1,181 @@
+        int: Number given by user.
+    """
+    while True:
+        number = input(message_input + "\n")
+        if not number.isdigit():
+            print("This is not a number, try again.")
+            continue
+        break
+    return int(number)
+
+
+def get_balance(customers_data, customer_id):
+    """Returns the given customer's balance as an integer.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+
+    Returns:
+        int: The given customer's balance.
+    """
+    balance = customers_data[customer_id][BALANCE_KEY]
+    return int(balance)
+
+
+def cash_withdrawal(customers_data, customer_id):
+    """Deducts the 'amount_to_withdrawal' from the customer's balance.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+    """
+    msg = "Enter the amount you want to withdrawal."
+    amount_to_withdrawal = get_a_number(msg)
+    curr_balance = get_balance(customers_data, customer_id)
+    new_balance = curr_balance - amount_to_withdrawal
+    customers_data[customer_id][BALANCE_KEY] = new_balance
+    print(f"withdrawal was successful {customer_id} now has {new_balance}$")
+
+
+def cash_deposit(customers_data, customer_id):
+    """Adds the 'amount_to_deposit' to the customer's balance.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+    """
+    amount_to_deposit = get_a_number("Enter the amount you want to deposit.")
+    new_balance = get_balance(customers_data, customer_id) + amount_to_deposit
+    customers_data[customer_id][BALANCE_KEY] = new_balance
+    print(f"withdrawal was successful {customer_id} now has {new_balance}$")
+
+
+def change_password(customers_data, customer_id):
+    """Changes the given customer's password.
+
+    Args:
+        customers_data (dict): Dictionary with the atm file's data.
+        customer_id (str): The specific customer id to work on.
+    """
+    new_pass = input("Enter a new password.\n")
+    customers_data[customer_id][PASS_KEY] = new_pass
+    print(f"Password was changed to {new_pass}")
+
+
+def get_data_from_file(filename):
+    """Gets bank details from file, arranges and returns them in dictionary.
+
+    Args:
+        filename (str): The path to the atm details file.
+
+    Returns (dict): Dictionary of all customer's data from the file.
+    """
+    if os.path.exists(filename):
+        with open(filename, "r") as atm_file:
+            customers = {}
+            for line in atm_file:
+                customer_id, the_pass, balance = line.split()
+                customer = {PASS_KEY: the_pass, BALANCE_KEY: balance}
+                customers[customer_id] = customer
+        return customers
+    else:
+        print(f"the file path : '{filename}' doesn't exist")
+        return -1
```

Comment:
In Python we don't return `-1` to indicate a problem, we raise exceptions
Just let the `open` method raise the exception if the file does not exist